lib/modules/datasource/packagist/index.ts

```diff
@@ -119,27 +118,29 @@ export class PackagistDatasource extends Datasource {
     return packagistFile;
   }
 
-  private static extractDepReleases(versions: RegistryFile): ReleaseResult {
-    const dep: ReleaseResult = { releases: [] };
-    // istanbul ignore if
-    if (!versions) {
-      return dep;
+  /* istanbul ignore next */
+  private static extractDepReleases(
+    composerReleases: unknown
+  ): ReleaseResult | null {
+    const parsedRecord =
+      schema.ComposerReleasesRecord.safeParse(composerReleases);
+    if (parsedRecord.success) {
+      logger.trace('Packagist: extracting releases from record');
+      return schema.extractReleaseResult(Object.values(parsedRecord.data));
     }
-    dep.releases = Object.keys(versions).map((version) => {
-      // TODO: fix function parameter type: `versions`
-      const release = (versions as any)[version];
-      const parsedVersion = release.version ?? version;
-      dep.homepage = release.homepage || dep.homepage;
-      if (release.source?.url) {
-        dep.sourceUrl = release.source.url;
-      }
-      return {
-        version: parsedVersion.replace(regEx(/^v/), ''),
-        gitRef: parsedVersion,
-        releaseTimestamp: release.time,
-      };
-    });
-    return dep;
+
+    const parsedArray =
+      schema.ComposerReleasesArray.safeParse(composerReleases);
+    if (parsedArray.success) {
+      logger.trace('Packagist: extracting releases from array');
+      return schema.extractReleaseResult(parsedArray.data);
+    }
+
+    logger.warn(
+      { composerReleases },
+      'Packagist: unknown format to extract from'
+    );
```

Comment:
How likely is this to happen? remember warnings appear in PRs, dashboards, etc

Reply:
Oh, that's something I didn't want. Let's change it to `debug` or `trace` then?